src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+    private static final int RESIZE_COEFFICIENT = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+    private int capacity;
+
+    public static class Node<K,V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        @Override
+        public int hashCode() {
+            int result = 17;
+            result = 31 * result + (key == null ? 0 : key.hashCode());
+            return result;
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (capacity * LOAD_FACTOR);
+        capacity = DEFAULT_CAPACITY;
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            size = 0;
+            reSize();
+        }
+        Node<K, V> newNode = new Node<>(hashKey(key), key, value, null);
+        Node<K, V> nodeInTable = table[getIndex(key)];
```

Comment:
Instead of calculating getIndex(key) each time you can create int indexBucket and use variable

Reply:
I think better to use separete method to calculate index, instead of reassign variable in each method, when it's need to calculate index(for example, in put method and getValue method)

Reply:
I was talking about put method specifically
you use getIndex(key) more than once
just create at the beginning of the method int indexBucket = getIndex(key);

Reply:
Agree